### PR TITLE
Clucker Quiz

### PR DESCRIPTION
**Pull request by ianam**

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+const app = express();
+app.set("view engine", "ejs");
+app.use(morgan("dev"));
+
+app.use(express.static(path.join(__dirname, "public")));
+
+app.use(express.urlencoded({ extended: true }));
+app.use(cookieParser());
+
+app.use((req, res, next) => {
```
Comment (opposite-bracket) [inline app.js]: I would put some comment on top of this describing what it's intended to do. The code however, is pretty clear and I understand that the code is designed to get the username from the cookie and put it within the scope of the request.

File: app.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    res.locals.username = "";
+
+    if (username) {
+        res.locals.username = username;
+    }
+
+    next();
+});
+
+app.get('/sign_in', (req, res) => {
```
Comment (opposite-bracket) [inline app.js]: I would move this into its own directory, you already organized the clucks controllers into a clucks.js file within your routes folder,  i would maintain that organization standart by placing this into a file related to what its intended to do.

File: app.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    res.render("sign_in");
+});
+
+app.get('/', (req, res) => {
+    res.redirect("clucks/");
+});
+
+const COOKIE_MAX_AGE = 1000 * 60 * 60 * 24; // 1 day
```
Comment (opposite-bracket) [inline app.js]: I usually try to organize my code as follows:

```
imports
var definition
logic
return
```

That said at the end of the day, once you work in a team, they will have best practices placed for you to follow. In this case is not a big deal.

**Comment by ianam** [comment]
Arturo, as advised I reorganized my code into separate routes for authentication actions! Thanks for the tip!

**Pull request merged**